src/common.ts

```diff
@@ -65,18 +65,15 @@ export function resolvePath(cwd: string, arg: string) {
   return path.resolve(path.isAbsolute(arg) ? arg : path.join(cwd, arg));
 }
 
-/**
- * Follow rust std::path::Path::join
- * The advantage is it can handle joining 2 absolute paths with common part
-
- * Rust: join("/a/b","/a/c") => "/a/b/c"
- * Deno. join("/a/b","/a/c") => "/a/b/a/c"
-**/
-export function rustJoin(path1: string, path2: string) {
-  const maybeCommon = path.common([path1, path2]);
-  if (!maybeCommon) return path.join(path1, path2);
-
-  return path.join(maybeCommon, path1.replace(maybeCommon, ""), path2.replace(maybeCommon, ""));
+/**  Calculates destination path
+ * destination should be a directory
+ * example:
+ *          destination: /dir/a
+ *          from       : /path/file
+ *          returns    : /dir/a/file
+ */
+export function calculateDestinationPath(destination: string, from: string) {
```

Comment:
This existing actually really confuses me (even before with rustJoin). It just using the basename of `from` doesn't seem exactly right, but I'm not sure. I want to investigate what sh actually does so I'm going to look into that before merging this, but will do that on another day.

Reply:
I found something to follow https://github.com/uutils/coreutils/blob/main/src/uu/cp/src/cp.rs#L1030

it calls https://github.com/uutils/coreutils/blob/main/src/uu/cp/src/cp.rs#L1703 which its documentation looks very similar to the above function

Reply:
I wasn't really thinking of the problem here. This looks good. Thanks!

I'm just thinking that I think I'll move calculateDestinationPath to the implementation of cp_mv.ts. It seems specific to that.